specs/dispute-game-interface.md

```diff
@@ -24,53 +25,188 @@ dispute games to exist. If multiple dispute games run in production, it gives
 a similar security model as having multiple protocol clients, as a bug in a
 single dispute game will not result in the bug becoming consensus.
 
-## Dispute Game Interface
+## Types
 
-The dispute game interface should be generic enough to allow it to work with any
-proof system. This means that it should work fault proofs, validity proofs or
-an attestation based proof system.
+For added context, we define a few types that are used in the following snippets.
 
 ```solidity
 /// @notice The type of proof system being used.
 enum GameType {
-    Fault,
-    Validity,
-    Attestation
+    /// @dev The game will use a `IDisputeGame` implementation that utilizes fault proofs.
+    FAULT,
+    /// @dev The game will use a `IDisputeGame` implementation that utilizes validity proofs.
+    VALIDITY,
+    /// @dev The game will use a `IDisputeGame` implementation that utilizes attestation proofs.
+    ATTESTATION
+}
+
+/// @notice The current status of the dispute game.
+enum GameStatus {
+    /// @dev The game is currently in progress, and has not been resolved.
+    IN_PROGRESS,
+    /// @dev The game has concluded, and the `rootClaim` was challenged successfully.
+    CHALLENGER_WINS,
+    /// @dev The game has concluded, and the `rootClaim` could not be contested.
+    DEFENDER_WINS
+}
+
+/// @notice A dedicated timestamp type.
+type Timestamp is uint64;
+
+/// @notice A `Claim` type represents a 32 byte hash or other unique identifier for a claim about
+///         a certain piece of information.
+/// @dev For the `FAULT` `GameType`, this will be a MPT root representing the state of
```

Comment:
Will this be a mpt root or will it be an output proposal? I think if its an mpt root then we just need the preimage of the output proposal

Reply:
For the fault proof game, it'll be the merklized state of the fault proof program's memory at the end of execution if I understand correctly.